src/realm/sync/instruction_applier.cpp

```diff
@@ -891,7 +891,7 @@ void InstructionApplier::operator()(const Instruction::SetInsert& instr)
                         bad_transaction_log("SetInsert: NULL in non-nullable set '%2.%1'", field_name, table_name);
                     }
 
-                    if (data_type == type_Mixed || value.get_type() == data_type) {
+                    if (data_type == type_Mixed || value.is_null() || value.get_type() == data_type) {
```

Comment:
Unlike baas the test server allows nullable sets, but it hit an assertion failure if you tried to add one to a non-Mixed set.